packages/nodes-base/nodes/Mattermost/v1/actions/router.ts

```diff
@@ -0,0 +1,53 @@
+import {
+	IExecuteFunctions,
+} from 'n8n-core';
+
+import {
+	INodeExecutionData, NodeApiError,
+} from 'n8n-workflow';
+
+import * as channel from './channel';
+import * as message from './message';
+import * as reaction from './reaction';
+import * as user from './user';
+
+export async function router(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+	const items = this.getInputData();
+	const operationResult: INodeExecutionData[] = [];
+
+	for (let i = 0; i < items.length; i++) {
+		const resource = this.getNodeParameter('resource', i) as string;
+		let operation = this.getNodeParameter('operation', i) as string;
```

Comment:
The return types from `getNodeParameter` can and probably should be narrowed down. If we add to `getNodeParameter` two overloads for `resource` and `operation`, which are always a `string`, then we can remove the two castings here and convert the casting on L29 to a typing, to improve type safety.